dragon.c

```diff
@@ -137,10 +141,58 @@ void drag_end(GtkWidget *widget, GdkDragContext *context, gpointer user_data) {
         if (action_str[0] == 'i')
             free(action_str);
     }
-    if (and_exit)
+
+    if (and_exit){
         gtk_main_quit();
+    }
+
+    if(individual_exit) {
+        if (uri_count == 0) {
+            gtk_main_quit();
+            return;
+        }
+
+        GtkWidget* button = find_child(vbox,dd->text);
```

Comment:
Space after the comma.
```suggestion
        GtkWidget* button = find_child(vbox, dd->text);
```